### PR TITLE
getAllReturnIdsForOrder using new runDBStmtLongListById

### PR DESCRIPTION
**Pull request by GTaenzer**
new method getAllReturnIdsForOrder uses new common "runDBStmt" method used twice so far...

### CODE REVIEW

File: src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java [hunk 1]
```diff
@@ -222,12 +222,55 @@ public Long runDBStmtLongById(String query, Long id, String resultColumnName)
     }
 
     @Override
-    public ArrayList<String> runDBStmtString(String query, String resultColumnName)
+    public List<Long> runDBStmtLongListById(String query, Long id, String resultColumnName)
```
Comment (mrc-mm) [inline src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java:225]: optional parameters should better be in the last positions

File: src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java [hunk 2]
```diff
@@ -222,12 +222,55 @@ public Long runDBStmtLongById(String query, Long id, String resultColumnName)
...
+            }
+        }
+
+        log.info("Got {} Long results for column '{}' from '{}' using id '{}'.", result.size(), resultColumnName, query, id);
+
+        return result;
+    }
+
+    @Override
+    public List<String> runDBStmtString(String query, String resultColumnName)
```
Comment (mrc-mm) [inline src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java:268]: why not just call    runDBStmtLongListById(query, null::Long , resultColumnName)

File: src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java [hunk 3]
```diff
@@ -2762,8 +2801,16 @@ public String getDispatchError(Long dispatchRef)
...
     {
-        return runDBStmtBoolean("SELECT \"showMaxReturnQuantity\" FROM \"PlatformConfigDO\"", "showMaxReturnQuantity",
-                        true);
+        return runDBStmtBoolean("SELECT \"showMaxReturnQuantity\" FROM \"PlatformConfigDO\"", "showMaxReturnQuantity", true);
+    }
+
+    @Override
+    public List<Long> getAllReturnIdsForOrder(OMSOrder order)
+    {
+        String query = "SELECT id FROM oms.\"ReturnDO\" WHERE \"orderRef\" in (?)";
```
Comment (mrc-mm) [inline src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java:2810]: WHERE \"orderRef\" = ?

Comment (mrc-mm) [inline src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java:2810]: maybe add "ORDER BY id" to ensure a stable list?

File: src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java [hunk 4]
```diff
@@ -2836,10 +2850,8 @@ public Long getReturnPositionQuantityReturned(Long returnPosRef)
             }
             if (resultSet.next())
             {
-                log.error("More than one quantityReturned found for the return position with the id '" + returnPosRef
-                                + "'!");
-                throw new RuntimeException("More than one quantityReturned found for the return position with the id '"
-                                + returnPosRef + "'!");
+                log.error("More than one quantityReturned found for the return position with the id '" + returnPosRef + "'!");
```
Comment (mrc-mm) [inline src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java:2853]: Is the exception not logged anyway?

File: src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java (hunk 2 above)
Comment (GTaenzer) [reply src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java:268]: minimal invasive - ArrayList is just not a return type...

File: src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java (hunk 1 above)
Comment (GTaenzer) [reply src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java:225]: Nothing optional here?

File: src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java (hunk 3 above)
Comment (GTaenzer) [reply src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java:2810]: done

File: src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java [hunk 5]
```diff
@@ -2826,7 +2840,7 @@ public Long getReturnPositionQuantityReturned(Long returnPosRef)
         String query = "SELECT \"quantityReturned\" FROM oms.\"ReturnPosDO\" WHERE \"id\" in (?)";
```
[the remark this answers is not in the document]
Comment (GTaenzer) [reply src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java:2840]: done

File: src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java (hunk 3 above)
Comment (GTaenzer) [reply src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java:2810]: done

File: src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java (hunk 4 above)
Comment (GTaenzer) [reply src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java:2853]: yeah, but logger and Exception are shown in different areas in the test output so it's nice to have the logger log the exception in the flow of the output...

File: src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java (hunk 1 above)
Comment (mrc-mm) [reply src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java:225]: I meant nullable parameter.

**Review by mrc-mm — APPROVED** [review]

File: src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java (hunk 1 above)
Comment (GTaenzer) [reply src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java:225]: I just wanted to be nice! ;-) (and accept nulls as well)

Comment (GTaenzer) [reply src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java:225]: But the method is called "...ById" and that what it really for....

**Pull request merged**